testutil/mock_calls.go

```diff
@@ -33,6 +33,12 @@ func (c *CallCounter) Get() map[string]int {
 	return maps.Clone(c.counts)
 }
 
+func (c *CallCounter) Reset() {
```

Comment:
Im failing to see where do we use this one

Reply:
[Here](https://github.com/hashicorp/nomad/pull/20532/files#diff-e90d51a406b0aa090a361afc3207f09c62da9d012d7e397e195e9f9e34958acfR361) so that we can use the `MountVolume` code path to setup an expected environment and then reset the calls so that the test assertion only lists the unmount/unpublish calls we expect.